### PR TITLE
[MB-13490] PPM closeout expenses detail page

### PR DESCRIPTION
**Pull request by duncan-truss**
## Summary

This PR adds a page component for the PPM Closeout Expenses page.

The form was already built so there wasn't much work to do there other than aligning our api field names with the field names in the form.  I added a margin to the missing weight info box that wasn't spaced like the designs.

I added the movingExpenses payload to the mtoShipment payload being returned by the API so that existing expenses would show on the review screen and that you could navigate to the edit screen not just the add new expense screen.  There is a follow up story to integrate calling the create/update moving expense endpoints, which I've left unfinished.  Similarly new uploads will not work because that is not yet integrated into the page component callback but will display on an existing expense.

Basic Cypress tests have been added to test filling out a new expense and editing an existing expense but without implementing uploads we can't yet submit the form.

[Deployed Storybook for PR](https://output.circle-artifacts.com/output/job/350db39c-d6c9-45ac-bf24-69f31a639568/artifacts/0/storybook/index.html?path=/story/customer-components-ppm-closeout-expenses-ppm-form--blank)

## Setup to Run Your Code

<details>
<summary>💻 You will need to use three separate terminals to test this locally.</summary>

##### Terminal 1

Populate the database and start the Go server locally.

```sh
make db_dev_e2e_populate && make server_run
```

##### Terminal 1

Start the UI locally.

```sh
make client_run
```

</details>

### Additional steps

1. Login to the customer app using the username `movingExpensePPM@ppm.approved`
2. On the home page click on the Upload PPM Documentation button for the existing approved PPM shipment
3. From the review page you can either click Add Expense to fill out the blank form or edit an existing expense.

## Verification Steps for Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Request review from a member of a different team.
- [ ] Have the Jira acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Frontend

- [ ] User facing changes have been reviewed by design.
- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
- [ ] There are no new console errors in the browser devtools
- [ ] There are no new console errors in the test output
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide#logging)
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/dev/contributing/backend/Backend-Programming-Guide/#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#zero-downtime-migrations)
- [ ] Have been communicated to #g-database
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/dev/contributing/database/Database-Migrations#secure-migrations)

## Screenshots

Screenshot 2022-09-29 at 11-04-19 my move mil
Screenshot 2022-09-29 at 11-04-46 my move mil
Screenshot 2022-09-29 at 11-07-16 my move mil
Screenshot 2022-09-29 at 11-08-13 my move mil
Screenshot 2022-09-29 at 11-08-39 my move mil
Screenshot 2022-09-29 at 11-08-54 my move mil

### CODE REVIEW

**Comment by rileyorr** [comment]
Hey @duncan-truss! I'm still familiarizing myself with the tool's component names and want to make sure I'm reviewing the right component you made margin updates to - is this the missing weight info box you're referring to?

Screen Shot 2022-09-29 at 2 12 47 PM

**Comment by duncan-truss** [comment]
> Hey @duncan-truss! I'm still familiarizing myself with the tool's component names and want to make sure I'm reviewing the right component you made margin updates to - is this the missing weight info box you're referring to?
> 
> Screen Shot 2022-09-29 at 2 12 47 PM

@rileyorr No worries, yes that is where I added the margin between the checkbox and the info box when the customer says they don't have the receipt.  That expenses form section was completed in a previous story so I didn't make any other tweaks.  I think it looked okay in Storybook before but sometimes there is different stylesheet precedence there compared to the live app.

Screen Shot 2022-09-29 at 2 47 13 PM

File: src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+const mockMTOShipmentId = '6b7a5769-4393-46fb-a4c4-d3f6ac7584c7';
+const mockExpenseId = 'ba29f5f5-0a51-4161-adaa-c568f5d5eab0';
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockPush,
+    replace: mockReplace,
+  }),
+  useParams: jest.fn(() => ({
+    moveId: 'cc03c553-d317-46af-8b2d-3c9f899f6451',
```
Comment (scottmries) [inline src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx:26]:
```suggestion
    moveId: mockMoveId,
```

File: src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx [hunk 2]
```diff
@@ -0,0 +1,166 @@
...
+    mtoShipmentId: '6b7a5769-4393-46fb-a4c4-d3f6ac7584c7',
```
Comment (scottmries) [inline src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx:27]:
```suggestion
    mtoShipmentId: mockMTOShipmentId,
```

File: src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+    await waitFor(() => {
+      expect(selectMTOShipmentById).toHaveBeenCalledWith(expect.anything(), mockMTOShipmentId);
+    });
+
+    expect(selectExpenseAndIndexById).toHaveBeenCalledWith(expect.anything(), mockMTOShipmentId, undefined);
+  });
+
+  it('loads the selected shipment and existing expense from redux', async () => {
+    useParams.mockImplementationOnce(() => ({
+      moveId: 'cc03c553-d317-46af-8b2d-3c9f899f6451',
```
Comment (scottmries) [inline src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx:88]:
```suggestion
      moveId: mockMoveId,
```

**Review by scottmries — APPROVED** [review]
I checked that
- the page loads
- the form is present
- the date fields appear when it's a SIT expense
- the fields validate.

I left just a couple non-blocking suggestions, but it LGTM otherwise.

**Review by rileyorr — APPROVED** [review]
@duncan-truss thank you, that explanation was very helpful! LGTM

File: src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx (hunk 1 above)
Comment (duncan-truss) [reply src/pages/MyMove/PPM/Closeout/Expenses/Expenses.test.jsx:26]: Let me try it, sometimes jest complains about using variables in mocks.  The `mock` prefix sometimes avoids that warning but other times I feel like there's still some initialization warning.

**Pull request merged**